test/helpers/utils.js

```diff
@@ -99,7 +100,32 @@ const giveEth = (toAccount, percentage, fromAccounts) => {
   });
 };
 
+const onlyWhitelisted = async (config, ifWhiteListed) => {
+  const { network, web3, accounts } = config;
+  const from = accounts[0];
+  if (network === 'ropstenGeth' || network === 'ropsten') {
+    if (
+      ![
+        '0xf1bcd758cb3d46d15afe4faef942adad36380148',
+        '0x2e4d8353d81b7e903c9e031dab3e9749e8ab69bc',
+      ].includes(from.toLowerCase())
+    ) {
+      throw new Error(
+        `${from} is not a whitelisted account for deploying to ropsten.`
+      );
+    }
+  } else if (network === 'develop' || network === 'test') {
+    if (!getNamedAccounts(web3).allAccounts.includes(from.toLowerCase())) {
+      throw new Error(
+        `${from} is not a whitelisted account for deploying to ${network}.`
+      );
+    }
+  }
+  return ifWhiteListed(config);
```

Comment:
How about just

```js
return config;
```

Reply:
I dont follow, Im passing a function which needs to be executed if the account is whitelisted, and then passing the config into that function

Reply:
use if statements. Which is better:

```js
const ifCondition = async (doSomethingConfig, doSomething) => {
  someCondition = ...;
  if (someCondition) {
    doSomething(doSomethingConfig);
  }
};
ifCondition("hello", console.log);
```

or

```
const isConditionTrue = () => {
  const someCondition = ...;
  return someCondition;
}
if (isConditionTrue) {
  console.log("hello");
}
```

You've written the first version, I'm suggesting the second.